libraries/AP_BattMonitor/AP_BattMonitor_SMBus_Maxell.cpp

```diff
@@ -54,6 +64,11 @@ void AP_BattMonitor_SMBus_Maxell::read()
 
 void AP_BattMonitor_SMBus_Maxell::timer()
 {
+	// _pec_confirmed set true after confirming if it support PEC
+    if (!_pec_confirmed) {
+        _pec_confirmed = get_pec_support();
+    }
```

Comment:
It's difficult to determine that the version is unknown.
It tries to get the correct data constantly from the battery when _pec_confirmed is false.
This can prevent to false recognition in case the initial data is incorrect.